tests/extra/job_management/test_job_management.py

```diff
@@ -720,6 +748,47 @@ def get_status(job_id, current_status):
         filled_running_start_time = final_df.iloc[0]["running_start_time"]
         assert isinstance(rfc3339.parse_datetime(filled_running_start_time), datetime.datetime)
 
+    def test_process_threadworker_updates(self, job_manager, tmp_path):
```

Comment:
this test seems to be interacting too much with internal implementation details of _JobManagerWorkerThreadPool to be a test about the job manager:
- `DummyTask` copies `_JobStartTask` behavior, why not use `_JobStartTask` directly?
- actively waiting for the futures to be done: that should be work for the job manager
- calling `_process_threadworker_updates`: that should be work for job manager too